### PR TITLE
fix(runtime): slot regressions from experimental slot fixes

### PR DESCRIPTION
**Pull request by tanner-reits**
## What is the current behavior?

A few changes made behind the `experimentalSlotFixes` flag were resulting in screenshot test failures in the Ionic Framework

GitHub Issue Number: N/A

## What is the new behavior?

- Fixes the conditional responsible for patching element methods for the pseudo shadow DOM when `experimentalSlotFixes` is enabled.
- Updated the `appendChild` patch to force a re-render of the patched element

See commit messages for more information

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Testing

**Automated tests**
- Stencil unit and e2e tests continue to pass.
- A build of Stencil containing these changes (and changes from the other "slot regressions" branches) was used in a manually triggered run of the Stencil nightly build and succeeded. Run results: https://github.com/ionic-team/ionic-framework/actions/runs/7280624127

### CODE REVIEW

File: src/runtime/dom-extras.ts [hunk 1]
```diff
@@ -84,10 +84,14 @@ export const patchSlotAppendChild = (HostElementPrototype: any) => {
     if (slotNode) {
       const slotChildNodes = getHostSlotChildNodes(slotNode, slotName);
       const appendAfter = slotChildNodes[slotChildNodes.length - 1];
-      appendAfter.parentNode.insertBefore(newChild, appendAfter.nextSibling);
+      const insertedNode = appendAfter.parentNode.insertBefore(newChild, appendAfter.nextSibling);
+
       // Check if there is fallback content that should be hidden
       updateFallbackSlotVisibility(this);
-      return;
+      // Fore a re-render of the host element
```
Comment (rwaskiewicz) [inline src/runtime/dom-extras.ts:91]: Non-blocking nit:
```suggestion
      // Force a re-render of the host element
```

File: src/runtime/dom-extras.ts [hunk 2]
```diff
@@ -84,10 +84,14 @@ export const patchSlotAppendChild = (HostElementPrototype: any) => {
...
+      forceUpdate(this);
```
Comment (rwaskiewicz) [inline src/runtime/dom-extras.ts:92]: Can you help me understand why we need to do force an update here?

File: test/karma/test-app/dom-reattach-clone/cmp-deep-slot.tsx [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { Component, h } from '@stencil/core';
 
 @Component({
   tag: 'dom-reattach-clone-deep-slot',
+  scoped: true,
```
Comment (rwaskiewicz) [inline test/karma/test-app/dom-reattach-clone/cmp-deep-slot.tsx:5]: For my own understanding, why do we need to add `scoped` to the components in this test suite?

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]

File: src/runtime/dom-extras.ts (hunk 2 above)
Comment (rwaskiewicz) [reply src/runtime/dom-extras.ts:92]: Note: I want a little additional clarification here, after reading the git commit msg

**Review by christian-bromann — APPROVED** [review]
Don't anything in addition to what @rwaskiewicz already pointed out 👍

File: test/karma/test-app/dom-reattach-clone/cmp-deep-slot.tsx (hunk 3 above)
Comment (tanner-reits) [reply test/karma/test-app/dom-reattach-clone/cmp-deep-slot.tsx:5]: The test uses the `cloneNode` method on a Stencil component, but with the changes to the bootstrap methods in this PR, we only patch native methods for `scoped` components when `experimentalSlotFixes` is enabled.

File: src/runtime/dom-extras.ts (hunk 2 above)
Comment (tanner-reits) [reply src/runtime/dom-extras.ts:92]: I ran into an issue when testing this out with Framework screenshot tests. For some components that had `appendChild` patched, the component would include the append child node after the call until something triggered a re-render of the component (this was as simple as hovering the component). This made it so the component would immediately re-render to reflect the changes and get the tests to pass

**Review by rwaskiewicz — APPROVED** [review]

**Pull request merged**